sdk/metric/internal/filter.go

```diff
@@ -31,15 +49,11 @@ type filter[N int64 | float64] struct {
 	seen map[attribute.Set]attribute.Set
 }
 
-// NewFilter wraps an Aggregator with an attribute filtering function.
-func NewFilter[N int64 | float64](agg Aggregator[N], fn attribute.Filter) Aggregator[N] {
-	if fn == nil {
-		return agg
-	}
+func newFilter[N int64 | float64](agg Aggregator[N], fn attribute.Filter) *filter[N] {
```

Comment:
this is only for non-precomputed filters, right?

Reply:
Right. I'm AFK but can add a comment when I get back.

Reply:
Added comment to docs.